backends/dpdk/dpdkAsmOpt.cpp

```diff
@@ -163,4 +163,41 @@ const IR::IndexedVector<IR::DpdkAsmStatement> *RemoveLabelAfterLabel::removeLabe
     }
     return new_l;
 }
+
+const IR::Node* RemoveUnusedMetadataFields::preorder(IR::DpdkAsmProgram *p) {
+    auto usedFieldsCollector = new CollectUsedMetadataField();
+    usedFieldsCollector->setCalledBy(this);
+    p->apply(*usedFieldsCollector);
+    auto used_fields = usedFieldsCollector->getUsedFields();
+
+    IR::IndexedVector<IR::DpdkStructType> usedStruct;
+    bool isMetadataStruct = false;
+    for (auto st : p->structType) {
+        if (!isMetadataStruct) {
+            for (auto anno : st->annotations->annotations) {
```

Comment:
Can st->annotations be null in any case? It would be better if it is checked for nullptr before continue.